### ISSUE TITLE
close_socket double free or corruption [May  2 09:42]

### ISSUE THREAD

**Issue by rabidt** [LuminariMUD/Luminari-Source#19]
```
May  2 09:42:05 :: Bad PW: Zackery [73.53.22.127]
May  2 09:42:06 :: Bad PW: Zackery [73.53.22.127]
SYSERR: Write to socket: Broken pipe
May  2 09:42:09 :: Losing descriptor without char.
May  2 09:42:09 :: Losing descriptor without char.
*** Error in `circle': double free or corruption (out): 0x000000000f457360 ***
```
```
Program terminated with signal 6, Aborted.
#0  0x00007f72f8a55a98 in raise () from /lib64/libc.so.6
Missing separate debuginfos, use: debuginfo-install MariaDB-shared-10.1.44-1.el7.centos.x86_64 bzip2-libs-1.0.6-13.el7.x86_64 expat-2.1.0-10.el7_3.x86_64 fontconfig-2.13.0-4.3.el7.x86_64 freetype-2.8-14.el7.x86_64 gd-2.0.35-26.el7.x86_64 glibc-2.22-3.fc23.x86_64 keyutils-libs-1.5.8-3.el7.x86_64 krb5-libs-1.15.1-37.el7_7.2.x86_64 libX11-1.6.7-2.el7.x86_64 libXau-1.0.8-2.1.el7.x86_64 libXpm-3.5.12-1.el7.x86_64 libcom_err-1.42.9-16.el7.x86_64 libgcc-5.1.1-4.fc23.x86_64 libjpeg-turbo-1.2.90-8.el7.x86_64 libpng-1.5.13-7.el7_2.x86_64 libselinux-2.5-14.1.el7.x86_64 libstdc++-5.1.1-4.fc23.x86_64 libuuid-2.23.2-61.el7_7.1.x86_64 libxcb-1.13-1.el7.x86_64 nss-softokn-freebl-3.44.0-8.el7_7.x86_64 openssl-libs-1.0.2k-19.el7.x86_64 pcre-8.32-17.el7.x86_64 zlib-1.2.8-9.fc23.x86_64
(gdb) bt
#0  0x00007f72f8a55a98 in raise () from /lib64/libc.so.6
#1  0x00007f72f8a5769a in abort () from /lib64/libc.so.6
#2  0x00007f72f8a98e1a in __libc_message () from /lib64/libc.so.6
#3  0x00007f72f8aa141a in _int_free () from /lib64/libc.so.6
#4  0x00007f72f8aa4bcc in free () from /lib64/libc.so.6
#5  0x0000000000573c66 in close_socket (d=0x12d45e60) at comm.c:2846
#6  0x0000000000576d96 in game_loop (local_mother_desc=3) at comm.c:1022
#7  0x0000000000404b2a in init_game (local_port=<optimized out>) at comm.c:587
#8  main (argc=<optimized out>, argv=<optimized out>) at comm.c:392
(gdb) f 5
#5  0x0000000000573c66 in close_socket (d=0x12d45e60) at comm.c:2846
warning: Source file is more recent than executable.
2846            free(d->history[cnt]);
(gdb) l
2841      if (d->history)
2842      {
2843        int cnt;
2844        for (cnt = 0; cnt < HISTORY_SIZE; cnt++)
2845          if (d->history[cnt])
2846            free(d->history[cnt]);
2847        free(d->history);
2848      }
2849
2850      if (d->showstr_head)
```

**Comment by moshehbenavraham** [comment]
@GickerLDS Is that a stock issue?

**Comment by Odoth** [comment]
Repeated with binary based on aaa617c5501dfdea4aec8ab9edf3944d6a17392f

```
May 12 20:28:58 :: SYSERR: Mob using '((vict)->player_specials->saved.pref)' at act.offensive.c:2362.
SYSERR: Write to socket: Connection reset by peer
May 12 20:29:01 :: Losing descriptor without char.
May 12 20:29:01 :: Losing descriptor without char.
*** Error in `circle': double free or corruption (out): 0x0000000017b94df0 ***

```
```
Core was generated by `circle -C3 4100'.
Program terminated with signal 6, Aborted.
#0  0x00007f72c6fb0a98 in raise () from /lib64/libc.so.6
Missing separate debuginfos, use: debuginfo-install MariaDB-shared-10.1.44-1.el7.centos.x86_64 bzip2-libs-1.0.6-13.el7.x86_64 expat-2.1.0-10.el7_3.x86_64 fontconfig-2.13.0-4.3.el7.x86_64 freetype-2.8-14.el7.x86_64 gd-2.0.35-26.el7.x86_64 glibc-2.22-3.fc23.x86_64 keyutils-libs-1.5.8-3.el7.x86_64 krb5-libs-1.15.1-37.el7_7.2.x86_64 libX11-1.6.7-2.el7.x86_64 libXau-1.0.8-2.1.el7.x86_64 libXpm-3.5.12-1.el7.x86_64 libcom_err-1.42.9-16.el7.x86_64 libgcc-5.1.1-4.fc23.x86_64 libjpeg-turbo-1.2.90-8.el7.x86_64 libpng-1.5.13-7.el7_2.x86_64 libselinux-2.5-14.1.el7.x86_64 libstdc++-5.1.1-4.fc23.x86_64 libuuid-2.23.2-61.el7_7.1.x86_64 libxcb-1.13-1.el7.x86_64 nss-softokn-freebl-3.44.0-8.el7_7.x86_64 openssl-libs-1.0.2k-19.el7.x86_64 pcre-8.32-17.el7.x86_64 zlib-1.2.8-9.fc23.x86_64
(gdb) bt
#0  0x00007f72c6fb0a98 in raise () from /lib64/libc.so.6
#1  0x00007f72c6fb269a in abort () from /lib64/libc.so.6
#2  0x00007f72c6ff3e1a in __libc_message () from /lib64/libc.so.6
#3  0x00007f72c6ffc41a in _int_free () from /lib64/libc.so.6
#4  0x00007f72c6fffbcc in free () from /lib64/libc.so.6
#5  0x00000000005556f6 in close_socket (d=0x18885180) at comm.c:2854
#6  0x00000000005588b6 in game_loop (local_mother_desc=3) at comm.c:1021
#7  0x0000000000404a4a in init_game (local_port=<optimized out>) at comm.c:587
#8  main (argc=<optimized out>, argv=<optimized out>) at comm.c:392
(gdb) f 5
#5  0x00000000005556f6 in close_socket (d=0x18885180) at comm.c:2854
warning: Source file is more recent than executable.
2854            free(d->history[cnt]);
(gdb) l
2849      if (d->history)
2850      {
2851        int cnt;
2852        for (cnt = 0; cnt < HISTORY_SIZE; cnt++)
2853          if (d->history[cnt])
2854            free(d->history[cnt]);
2855        free(d->history);
2856      }
2857
2858      if (d->showstr_head)
```

**Comment by Odoth** [comment]
Should be fixed by 5418e212142b09849e2741dfe3ea1f3fa252a939
